pkg/setup/id_reserver_test.go

```diff
@@ -80,10 +80,10 @@ func TestIdReserver_ReserveIDs(t *testing.T) {
 	timeout := time.Second
 
 	type testCase struct {
-		testName string                                  // test name
-		routers  map[cipher.PubKey]*mockGatewayForDialer // arrange: map of mock router gateways
-		paths    [][]routing.Hop                         // arrange: idReserver input
-		expErr   error                                   // assert: expected error
+		testName string                        // test name
+		routers  map[cipher.PubKey]interface{} // arrange: map of mock router gateways
```

Comment:
Why would `*mockGatewayForDialer` not be better here?

Reply:
@nkryuchkov because we may have different implementations for the RPC Gateway for future tests.

In fact, as I was writing these tests, there was a time point when I had multiple implementations, but I deemed them not necessary.

I think we should leave it as it, just in case for possible future changes.

Reply:
@evanlinjin What do you think about creating a common interface for RPC Gateway implementations and use it in this `map`?